core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -53,6 +53,7 @@ private[spark] class NettyBlockTransferService(
   private val serializer = new JavaSerializer(conf)
   private val authEnabled = securityManager.isAuthenticationEnabled()
   private val transportConf = SparkTransportConf.fromSparkConf(conf, "shuffle", numCores)
+  private val blockFetchers = collection.mutable.HashSet[OneForOneBlockFetcher]()
```

Comment:
this is not thread safe. And it's a memory leak because it won't be cleaned until the executor is shutting down.